piksi_tools/bootload.py

```diff
@@ -58,11 +58,13 @@ def __exit__(self, *args):
       self.stop()
 
   def stop(self):
+    """ Remove Bootloader instance callbacks from serial link. """
     self.stopped = True
     self.link.remove_callback(self._deprecated_callback, SBP_MSG_BOOTLOADER_HANDSHAKE_DEPRECATED)
     self.link.remove_callback(self._handshake_callback, SBP_MSG_BOOTLOADER_HANDSHAKE_DEVICE)
 
   def _deprecated_callback(self, sbp_msg):
+    """ Bootloader handshake for deprecated message ID. """
     if len(sbp_msg.payload)==1 and struct.unpack('B', sbp_msg.payload[0])==0:
```

Comment:
@mookerji I think this will only work for the most recent version of the message at the moment, as there are fields in the older version that aren't present in the message from the device with the old bootloader.

```
Traceback (most recent call last):
  File "/Users/colin/miniconda/lib/python2.7/site-packages/sbp-0.45.post8_g805031f-py2.7.egg/sbp/client/handler.py", line 140, in run
    self.call(msg)
  File "/Users/colin/miniconda/lib/python2.7/site-packages/sbp-0.45.post8_g805031f-py2.7.egg/sbp/client/handler.py", line 217, in call
    callback(msg)
  File "./bootload.py", line 68, in _deprecated_callback
    bh = MsgBootloaderHandshakeDevice(sbp_msg)
  File "/Users/colin/miniconda/lib/python2.7/site-packages/sbp-0.45.post8_g805031f-py2.7.egg/sbp/bootload.py", line 97, in __init__
    self.from_binary(sbp.payload)
  File "/Users/colin/miniconda/lib/python2.7/site-packages/sbp-0.45.post8_g805031f-py2.7.egg/sbp/bootload.py", line 113, in from_binary
    p = MsgBootloaderHandshakeDevice._parser.parse(d)
  File "/usr/local/lib/python2.7/site-packages/construct/core.py", line 188, in parse
    return self.parse_stream(BytesIO(data))
  File "/usr/local/lib/python2.7/site-packages/construct/core.py", line 198, in parse_stream
    return self._parse(stream, Container())
  File "/usr/local/lib/python2.7/site-packages/construct/core.py", line 670, in _parse
    subobj = sc._parse(stream, context)
  File "/usr/local/lib/python2.7/site-packages/construct/core.py", line 365, in _parse
    raise FieldError(sys.exc_info()[1])
FieldError: expected 4, found 1
```

We could add a message class for the old message?